torch/csrc/jit/register_c10_ops.cpp

```diff
@@ -31,6 +34,57 @@ Operator createOperatorFromC10(const c10::OperatorHandle& op) {
         }
       }
 
+      if (jit::tracer::isTracing()) {
+        auto symbol = Symbol::fromQualString(op.schema().name());
+        const auto& graph = tracer::getTracingState()->graph;
+        node = graph->create(symbol, 0);
+        const auto& args = op.schema().arguments();
+        int i = 0;
+        for (auto iter = stack.end() - input_size; iter != stack.end();
+             ++iter, ++i) {
+          // TODO we need to refactor graph APIs (e.g., addInputs)
+          // appropriately; after that, we can get rid of the giant if-else
+          // block we will clean this tech debt together in the following PRs
+          if (iter->isTensor()) {
```

Comment:
I think it's more sane to branch here on the dtype of argument in schema instead of trying to deduce the dtype from ivalue. (you already check it below in the optional branch)

Reply:
Sure, but the type system itself is not very user friendly...